iceoryx_utils/include/iceoryx_utils/internal/cxx/storable_function.inl

```diff
@@ -0,0 +1,250 @@
+    : m_vtable(other.m_vtable)
+{
+    m_vtable.copy(other, *this);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::storable_function(storable_function&& other) noexcept
+    : m_vtable(other.m_vtable)
+{
+    m_vtable.move(other, *this);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>&
+storable_function<S, signature<ReturnType, Args...>>::operator=(const storable_function& rhs) noexcept
+{
+    if (&rhs != this)
+    {
+        // note: src vtable is needed for destroy, then changed to src vtable
+        m_vtable.destroy(*this);
+        m_function = nullptr; // only needed when the src has no object
+        m_vtable = rhs.m_vtable;
+        m_vtable.copy(rhs, *this);
+    }
+
+    return *this;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>&
+storable_function<S, signature<ReturnType, Args...>>::operator=(storable_function&& rhs) noexcept
+{
+    if (&rhs != this)
+    {
+        // note: dest (this) vtable is needed for destroy, then changed to source (rhs) vtable
+        m_vtable.destroy(*this);
+        m_function = nullptr; // only needed when the source has no object
+        m_vtable = rhs.m_vtable;
+        m_vtable.move(rhs, *this);
+    }
+
+    return *this;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::~storable_function() noexcept
+{
+    m_vtable.destroy(*this);
+}
+
+
+template <typename S, typename ReturnType, typename... Args>
+ReturnType storable_function<S, signature<ReturnType, Args...>>::operator()(Args... args)
+{
+    auto r = m_function(std::forward<Args>(args)...);
+    return r;
+}
+
+template <typename S, typename ReturnType, typename... Args>
+storable_function<S, signature<ReturnType, Args...>>::operator bool() noexcept
+{
+    return m_function.operator bool();
+}
+
+template <typename S, typename ReturnType, typename... Args>
+void storable_function<S, signature<ReturnType, Args...>>::swap(storable_function& f) noexcept
+{
+    storable_function tmp = std::move(f);
+    f = std::move(*this);
+    *this = std::move(tmp);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+void storable_function<S, signature<ReturnType, Args...>>::swap(storable_function& f, storable_function& g) noexcept
+{
+    storable_function tmp = std::move(f);
+    f = std::move(g);
+    g = std::move(tmp);
+}
+
+template <typename S, typename ReturnType, typename... Args>
+template <typename Functor, typename>
+void storable_function<S, signature<ReturnType, Args...>>::storeFunctor(const Functor& functor) noexcept
+{
+    ///@todo we may need something more to get the correct type in general...
+    using StoredType = typename std::remove_reference<Functor>::type;
+    auto p = m_storage.template allocate<StoredType>();
```

Comment:
Whats `p` ? `ptr` ?

Reply:
It is a ptr, yes. Implied by allocate (maybe). Personally I prefer to keep variable names with very small scope short unless something magical happens which can be explained by better names.

Reply:
I guess kind of nitpicky but I would argue for using conventional names like `ptr` when possible.
When I see `ptr` I have to spend 0 brain power to immediately understand what's happening.

Reply:
Throwing `pointer` in the ring. Simple, precise, straightforward. Opinions?

Reply:
`ptr` is my favourite because it's in the middle between `p` and `ptr` and it is common sense (type ptr in the search field of Github).

Reply:
pointer is to long, I can call it ptr. Still, I personally am a fan of short variable names in those kind of local scopes and it is fairly obvious that an allocation will provide one.
I do not like overly long degenerate names for no good reason.